editor/src/clj/editor/material.clj

```diff
@@ -145,14 +146,41 @@
             :user-data {:material-desc-with-build-resources material-desc-with-build-resources}
             :deps dep-build-targets})])))
 
+"A resource namespace is the first literal up until the first dot in a resource binding.
+For example, if we have a uniform buffer with some nested data types:
+
+struct MyMaterial {
+  vec4 diffuse;
+  vec4 specular;
+};
+
+uniform my_uniforms {
+  MyMaterial material;
+};
+
+When crosscompiled to SM120 (which is used by the editor), we will get two uniforms:
+_<id>.material.diffuse
+_<id>.material.specular
+
+To be able to map this in a material constant, we need to strip the namespace from the
+reflected data when the shader is created (see editor.gl.shader:make-shader-program) since
+there is no way a user can know what the generated id will be for older shaders.
+"
+(defn- resource-binding-namespaces [^SPIRVReflector reflector]
+  ;; SSBOs will need the same remapping as UBOs, but since we don't support
+  ;; them in the editor, we don't gather their namespaces here.
+  (mapv (fn [ubo] (str "_" (.id ubo))) (.getUBOs reflector)))
```

Comment:
This would use reflection, since you're not type-tagging the `ubo` argument. Also, this would read better if it was split across multiple lines.